LoginClient/LoginClientForm.cs

```diff
@@ -0,0 +1,230 @@
+{
+    public partial class LoginClientForm : Form
+    {
+        private PipeClient pipeClient;
+
+        public LoginClientForm()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// This method sets up the program when it first loads.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void LoginClientForm_Load(object sender, EventArgs e)
+        {
+            CreateNewPipeClient();
+        }
+
+        /// <summary>
+        /// Create a new pipeClient and set all the handlers for the delegates.
+        /// </summary>
+        private void CreateNewPipeClient()
+        {
+            if (pipeClient != null)
+            {
+                pipeClient.MessageRecieved -= pipeClient_MessageRecieved;
+                pipeClient.ServerDisconnected -= pipeClient_ServerDisconnected;
+                pipeClient.ConnectedToServer -= pipeClient_ConnectedToServer;
+            }
+
+            pipeClient = new PipeClient();
+            pipeClient.MessageRecieved += pipeClient_MessageRecieved;
+            pipeClient.ServerDisconnected += pipeClient_ServerDisconnected;
+            pipeClient.ConnectedToServer += pipeClient_ConnectedToServer;
+
+            UpdateForm();
+        }
+
+        /// <summary>
+        /// Methods to allow the pipeClient to use the methods on the form.
+        /// </summary>
+        #region HandlerMethods
+
+        private void pipeClient_ConnectedToServer()
+        {
+            Invoke(new PipeClient.ConnectedToServerHandler(UpdateForm));
+        }
+
+        private void pipeClient_MessageRecieved(byte[] message)
+        {
+            Invoke(new PipeClient.MessageRecievedHandler(DisplayRecievedMessage),
+                new object[] { message });
+        }
+
+        private void pipeClient_ServerDisconnected()
+        {
+            Invoke(new PipeClient.ServerDisconnectedHandler(UpdateForm));
+        }
+        #endregion
+
+        /// <summary>
+        /// Add the message onto a newline of the message log.
+        /// </summary>
+        /// <param name="message"></param>
+        private void DisplayRecievedMessage(byte[] message)
+        {
+            Console.WriteLine(Thread.CurrentThread.ManagedThreadId + "in DisplayRecievedMessage");
+            ASCIIEncoding encoder = new ASCIIEncoding();
+            string str = encoder.GetString(message, 0, message.Length);
+
+            if (str == "close")
+            {
+                //MessageLogTB.Text += str + "\r\n";
+                pipeClient.Disconnect();
+                CreateNewPipeClient();
+            }
+            MessageLogTB.Text += str + "\r\n";
+        }
+
+        /// <summary>
+        /// Update the enabled property of the buttons on the form.
+        /// Then resets the focus.
+        /// </summary>
+        private void UpdateForm()
+        {
+            bool connected = pipeClient.IsConnected();
```

Comment:
use constants here, it will help to make sure you dont override the values later by accident
const bool = pipeClient......
often bugs for new programmers happen in if statements where
if (variable = true){}  
will assign and pass.
while that would fail, if you used const.